### PR TITLE
Constructor refactor

### PR DESCRIPTION
**Pull request by jgallagher59701**
Updates for Constructor before fixes to the read() method.

### CODE REVIEW

**Comment by jgallagher59701** [comment]
NB: The BES passes all of its tests on my intel mac when using these changes.

File: Constructor.cc [hunk 1]
```diff
@@ -428,39 +334,40 @@ Constructor::add_var_nocopy(BaseType *bt, Part)
...
+ * remove all variables in this Constructor with name 'n' while the old
+ * version would just remove the first variable.
+ * @param n name of the variable to remove
+ */
 void
 Constructor::del_var(const string &n)
 {
-	// TODO remove_if? find_if?
-    for (Vars_iter i = d_vars.begin(); i != d_vars.end(); i++) {
+    for (auto i = d_vars.begin(); i != d_vars.end(); i++) {
```
Comment (ndp-opendap) [inline Constructor.cc:352]: maybe "var" instead of "i"

File: Constructor.cc [hunk 2]
```diff
@@ -828,74 +686,68 @@ Constructor::is_linear()
...
+
             AttrTable *bt_attr_table = new AttrTable(bt->get_attr_table());
             bt_attr_table->set_name(bt->name());
             string type_name = bt->type_name();
-            if(bt->is_vector_type()){
+
+            if (bt->is_vector_type()) {
                 Array *array = dynamic_cast <Array *>(bt);
-                if(array){
+                if (array) {    // This is always true - only an Array is_vector_type(). jhrg 4/25/22
```
Comment (ndp-opendap) [inline Constructor.cc:728]: If it's always true then why continue testing it?

**Review by ndp-opendap — APPROVED** [review]
It looks good, modulo my complete in ability to understand the lambda function thing and my nit-picky style comments. When the tests past you must :shipit:

File: Constructor.cc (hunk 2 above)
Comment (jgallagher59701) [reply Constructor.cc:728]: I was hesitant to change that code in this PR.

File: Constructor.h [hunk 3]
```diff
@@ -51,24 +51,38 @@ class Constructor: public BaseType
...
+        BaseType::operator=(rhs);
+        m_duplicate(rhs);
+        return *this;
+    }
 
-    Constructor &operator=(const Constructor &rhs);
     void transform_to_dap4(D4Group *root, Constructor *dest) override;
 
-    std::string FQN() const override;
+    std::string FQN() const override ;
```
[the remark this answers is not in the document]
Comment (jgallagher59701) [reply Constructor.h:85]: Nope. I saw that too, just as the thing went in.

When I get time, I'm going to look into a hook using astyle tied to git commits. But not right now.

**Pull request merged**